src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -82,13 +95,22 @@ public Task<ActionResult<IEnumerable<Models.ProcessIdentifier>>> GetProcesses()
         /// <summary>
         /// Get information about the specified process.
         /// </summary>
-        /// <param name="processKey">Value used to identify the target process, either the process ID, the runtime instance cookie, or process name.</param>
-        [HttpGet("processes/{processKey}", Name = nameof(GetProcessInfo))]
+        /// <param name="pid">Process ID used to identify the target process.</param>
+        /// <param name="uid">The Runtime instance cookie used to identify the target process.</param>
+        /// <param name="name">Process name used to identify the target process.</param>
+        [HttpGet("process", Name = nameof(GetProcessInfo))]
         [ProducesWithProblemDetails(ContentTypes.ApplicationJson)]
         [ProducesResponseType(typeof(Models.ProcessInfo), StatusCodes.Status200OK)]
         public Task<ActionResult<Models.ProcessInfo>> GetProcessInfo(
-            ProcessKey processKey)
+            [FromQuery]
+            int? pid = null,
+            [FromQuery]
+            Guid? uid = null,
+            [FromQuery]
+            string name = null)
         {
+            ProcessKey? processKey = (pid == null && uid == null && name == null) ? null : new ProcessKey(pid, uid, name);
```

Comment:
While this is somewhat simple, I would suggest refactoring this into a method because (1) it is used multiple times, and (2) it'll be much easier to update when we add other filter criteria (e.g. entrypoint name).